internal/decryption/keyproviders/azure.go

```diff
@@ -83,6 +84,14 @@ func (ki *AzureKeyIdentifier) ValidateCredentials() error {
 	return err
 }
 
-func (ki *AzureKeyIdentifier) DecryptKey(_ []byte) ([]byte, error) {
-	return nil, errors.New("not implemented")
+func (ki *AzureKeyIdentifier) DecryptKey(key []byte) ([]byte, error) {
+	client, err := crypto.NewClient(ki.KeyVaultEndpoint+"/keys/"+ki.KeyName+"/"+ki.KeyVersion, ki.credentials, nil)
+	if err != nil {
+		return nil, err
+	}
+	r, err := client.Decrypt(context.Background(), crypto.EncryptionAlgRSAOAEP, key, &crypto.DecryptOptions{})
```

Comment:
the algorithm is hardcoded for now, we should check it again with server/atlas teams when this is implemented

Reply:
link to a ticket for the follow up please